lib/pyszn/injection.py

```diff
@@ -150,14 +154,46 @@ def parse_attribute_injection(
                     result[filename]['environment'][attribute] = value
             except KeyError:
                 pass
+
+            # Get the topology for the filename and if it doesn't exist in the cache
```

Comment:
I am feeling that maybe this piece of code could have its own private method `_load_topo`

Reply:
Great idea, it will make things look cleaner and clearer.